cmd/kubectl-gadget/utils/k8s.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/kubectl-gadget/configflags"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/k8sutil"
+)
+
+func GetRunningGadgetNamespace() (string, bool) {
+	client, err := k8sutil.NewClientsetFromConfigFlags(configflags.KubernetesConfigFlags)
+	if err != nil {
+		log.Warnf("error creating k8s client: %s", err)
+		return "", false
+	}
+
+	nslist, err := client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Warnf("error listing namespaces: %s", err)
+		return "", false
+	}
+
+	gadgetNamespaces := make([]string, 0)
+	for _, ns := range nslist.Items {
+		daemonSet, err := client.AppsV1().DaemonSets(ns.Name).Get(context.TODO(), "gadget", metav1.GetOptions{})
+		if err != nil {
+			continue
+		}
+		if label, ok := daemonSet.Labels["k8s-app"]; ok {
+			if label == "gadget" {
+				gadgetNamespaces = append(gadgetNamespaces, ns.Name)
+			}
+		}
+	}
```

Comment:
What about listing daemon sets across all namespaces filtering by name and label?

```suggestion
	opts := metav1.ListOptions{
		FieldSelector: "metadata.name=gadget",
		LabelSelector: "k8s-app=gadget",
	}
	daemonSet, err := client.AppsV1().DaemonSets("").List(context.TODO(), opts)
	if err != nil {
		return "", false
	}

	for _, ds := range daemonSet.Items {
		gadgetNamespaces = append(gadgetNamespaces, ds.Namespace)
	}
```

Reply:
Didn't know that we could use `""` as the namespace. Works perfectly, thanks